### PR TITLE
Language.distill: copy both reference and predicted

### PR DESCRIPTION
**Pull request by danieldk**
## Description

In distillation we also modify the teacher docs (e.g. in tok2vec components), so we need to copy both the reference and predicted doc.

Problem caught by @shadeMe.

### Types of change

Bugfix

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/language.py [hunk 1]
```diff
@@ -2328,13 +2328,18 @@ def restore(self) -> None:
         self[:] = []
 
 
-def _copy_examples(examples: Iterable[Example]) -> List[Example]:
+def _copy_examples(
+    examples: Iterable[Example], copy_x: bool = True, copy_y: bool = False
```
Comment (shadeMe) [inline spacy/language.py:2332]: Can we make the new parameters keyword-only (if only to eliminate one potential source of confusion)?

File: spacy/language.py [hunk 2]
```diff
@@ -2328,13 +2328,18 @@ def restore(self) -> None:
...
     """Make a copy of a batch of examples, copying the predicted Doc as well.
     This is used in contexts where we need to take ownership of the examples
     so that they can be mutated, for instance during Language.evaluate and
     Language.update.
     """
-    return [Example(eg.x.copy(), eg.y) for eg in examples]
+    return [
+        Example(eg.x.copy() if copy_x else eg.x, eg.y.copy() if copy_y else eg.y)
+        for eg in examples
+    ]
```
Comment (shadeMe) [inline spacy/language.py:2342]: I wonder if we should emit a warning here if we do a no-op, i.e., both `copy_x` and `copy_y` are `False`. I can't think of a reason why the user would call the function at all in such a case, but it feels weird to have a function that basically goes `def do_something(really_do_something: bool = True):`?

File: spacy/language.py (hunk 1 above)
Comment (danieldk) [reply spacy/language.py]: Dones.

File: spacy/language.py (hunk 2 above)
Comment (danieldk) [reply spacy/language.py:2342]: It's an internal function. Besides that, I don't think it's a no-op since it clears things like cached alignments.

**Pull request merged**